File: apple/Sources/ReactNativeWatchOS/Modifiers/SearchSuggestions.swift
import SwiftUI

/// SwiftUI `.searchSuggestions { … }`.
///
/// Supplies the suggestion views for a searchable context. The ViewBuilder
/// overload is available on watchOS 9.0+, which is this bridge's deployment
/// target, so the modifier is applied unconditionally — no `#available`
/// guard is needed (one would only emit an "always true" warning here).
///
/// The `content` param is a hoisted `__ModifierContent` slot resolved through
/// `ctx`. When the slot is absent the view is returned unchanged. Note the
/// suggestions only render when the view is also marked searchable; on a
/// non-searchable view SwiftUI treats this as an inert pass-through.
enum RNWSearchSuggestionsModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("searchSuggestions") { view, params, ctx in
            guard let body = ctx.content(params.string("content")) else { return view }
            return AnyView(view.searchSuggestions { body })
        }
    }
}

File: src/modifiers/searchSuggestions.ts
import type { ReactNode } from 'react';
import { createModifier } from './createModifier';

export interface SearchSuggestionsParams {
  /// The suggestion views shown beneath a searchable field. Each child may
  /// carry a `searchCompletion(...)` modifier so tapping it fills the search
  /// field. Hoisted across the bridge as modifier content.
  content: ReactNode;
}

/// SwiftUI `.searchSuggestions { … }`.
///
/// Supplies the suggestion views for a searchable context. Available on
/// watchOS 9+ (the bridge deployment target); the native side still gates
/// defensively and falls back to the unmodified view on older systems.
///
/// The suggestions only render when this view is also marked searchable; on
/// a non-searchable view the modifier is an inert pass-through.
export function searchSuggestions(params: SearchSuggestionsParams) {
  return createModifier('searchSuggestions', params);
}
